lib/features/support/interactive_cli.rb

```diff
@@ -0,0 +1,117 @@
+require 'open3'
+
+# Encapsulates a terminal session, retaining state and input streams for interactive tests
+class InteractiveCLI
+  # @!attribute [r] parsed_output
+  #   @return [Array] An array of output strings received from the terminals STDOUT pipe
+  attr_reader :parsed_output
+
+  # @!attribute [r] parsed_error
+  #   @return [Array] An array of error strings received from the terminals STDERR pipe
+  attr_reader :parsed_errors
+
+  # @!attribute [r] last_exit_code
+  #   @return [Number] The exit code of the last terminal command
+  attr_reader :last_exit_code
+
+  # @!attribute [r] pid
+  #   @return [Number] The PID of the running terminal
+  attr_reader :pid
+
+  # @!attribute [r] current_buffer
+  #   @return [String] A string representation of the current output present in the terminal
+  attr_reader :current_buffer
+
+  # Creates an InteractiveCLI instance
+  #
+  # @param environment [Hash] A hash of environment variables
+  # @param terminal [String] A path to the terminal to run, defaults to `/bin/sh`
+  # @param terminal [String] The stop command, defaults to `exit`
+  def initialize(environment = {}, terminal = '/bin/sh', stop_command = 'exit')
+    @env = environment
+    @parsed_output = []
+    @parsed_errors = []
+    @current_buffer = ''
+    @last_exit_code = nil
+    @in_stream = nil
+    @pid = nil
+    @running = false
+    @stop_command = stop_command
+    start_threaded_terminal(terminal)
+  end
+
+  # Attempts to stop the terminal using the preset command
+  #
+  # @return [Boolean] true if successful, false otherwise
+  def stop
+    run_command(@stop_command)
```

Comment:
Does the Open3 thread exit after this is run? If so it might be worth waiting for that to happen and also the thread that's created by `start_threaded_terminal`, e.g.

```suggestion
    run_command(@stop_command)
    @thread.join(10)
```